core/src/HIP/Kokkos_HIP_Space.hpp

```diff
@@ -93,10 +101,15 @@ class HIPSpace {
                   const size_t arg_logical_size = 0) const;
 
  private:
-  void* impl_allocate(const hipStream_t stream, const char* arg_label,
+  void* impl_allocate(const HIP& exec_space, const char* arg_label,
                       const size_t arg_alloc_size,
-                      const size_t arg_logical_size,
-                      bool stream_sync_only) const;
+                      const size_t arg_logical_size = 0,
+                      const Kokkos::Tools::SpaceHandle =
+                          Kokkos::Tools::make_space_handle(name())) const;
```

Comment:
Why is it better to create the space handle here instead of locally when required? Is this mostly to match the `Cuda` implementation?

Reply:
Yeah, I wanted to match Cuda, but I can change that.